code-challenges.test.js

```diff
@@ -60,8 +98,197 @@ var hand2 = [5, 5, 3, 3, 4]
 var hand3 = [5, 5, 5, 5, 4]
 // Expected output: false
 
+describe("When isFullHouse is given an array of 5 numbers,", () => {
+
+    it("returns a boolean value determining if the array is a full house or not.", () => {
 
+        const actResult = isFullHouse(hand1);
+        
+        expect(actResult).toEqual(true);
+    })
+    it("returns a boolean value determining if the array is a full house or not.", () => {
 
+        const actResult = isFullHouse(hand2);
+        
+        expect(actResult).toEqual(false);
+    })
+    it("returns a boolean value determining if the array is a full house or not.", () => {
 
+        const actResult = isFullHouse(hand3);
+        
+        expect(actResult).toEqual(false);
+    })
+})
 
 // b) Create the function that makes the test pass.
+
+// Deven + Sarah discussion on how to go about this problem:
+// check for unique values and the count of the values 
+// store the cards in an object (key:value pair)
+// find two unique values and check to see if there are multiple of them 
+// keeping track of the characters and tracking them: update the value within the object
+
+const isFullHouse = (array) => {
+
+    if(array.length !== 5){
+        return "Please give me a 5 card hand to evaluate, no less no more."
+    }
+    // make sure given array is correct length for a poker hand
+
+    let count = {};
+    // object to store the counts of each occurring element
+
+    for(let i = 0; i < array.length; i++){
+        // iterate through all of the numbers in the array
+        let current = array[i];
+        // store the current number in a variable called 'current'
+        count[current] = count[current] ? count[current] + 1 : 1;
+        // place the current inside the object, then check if the current is seen in the object and add 1 to it for however many times it appeared. If it is not seen in the object again, keep the value of 1 (giving it an occurence of only 1)
+    }
+    
+    if(Object.values(count)[0] == 2 && Object.values(count)[1] == 3){
+        return true;
+
+    } else if(Object.values(count)[0] == 3 && Object.values(count)[1] == 2){
+        return true;
+
+    } else {
+        return false;
+
+    }
```

Comment:
because conditionals already return a boolean, explicit return of the booleans aren't necessary. One possible refactor of this could be:
```suggestion
    return Object.values(count)[0] == 2 && Object.values(count)[1] == 3 || Object.values(count)[0] == 3 && Object.values(count)[1] == 2;
```
If neither of these conditionals are true, false will be returned. Great job on the logic!